Packs/CiscoFirepower/Integrations/CiscoFirepower/CiscoFirepower.py

```diff
@@ -1,111 +1,527 @@
+            """
+            is_automatic: bool = limit is not None and limit > 0
+            is_manual: bool = (page is not None and page > 0) or (page_size is not None and page_size > 0)
+
+            if all((is_manual, is_automatic)):
+                raise ValueError('page or page_size can not be entered with limit.')
+
+            # Automatic Pagination
+            if is_automatic:
+                remaining_items = limit
+                offset = None
+
+            # Manual Pagination
+            elif is_manual:
+                page = page or 1
+                page_size = page_size or default_limit
+
+                remaining_items = page_size
+                offset = (page - 1) * page_size + (0 if start_count_from_zero else 1)
+
+            # No Pagination
+            else:
+                remaining_items = default_limit
+                offset = None
+
+            # API only supports limit parameter.
+            if not has_offset:
+                if has_limit:
+                    raw_response = func(
+                        self,
+                        limit=remaining_items if remaining_items <= api_limit else api_limit,  # type: ignore[operator]
+                        *args,
+                        **kwarg
+                    )
+
+                else:
+                    raw_response = func(
+                        self,
+                        *args,
+                        **kwarg
+                    )
+
+                items = raw_response
+
+                if items_key_path:
+                    items = dict_safe_get(items, items_key_path)
+
+                if is_automatic:
+                    items = items[:remaining_items]
+
+                else:  # is_manual
+                    stop = page * remaining_items  # type: ignore[operator]
+                    items = items[offset:stop]
+
+                return items, raw_response
+
+            raw_items: list[dict[str, Any]] = []
+            raw_responses: list[dict[str, Any]] = []
+
+            # Keep calling the API until the required amount of items have been met.
+            while remaining_items > 0:  # type: ignore[operator]
+                raw_response = func(
+                    self,
+                    limit=remaining_items if remaining_items <= api_limit else api_limit,  # type: ignore[operator]
+                    offset=offset,
+                    *args,
+                    **kwarg
+                )
+
+                raw_item = raw_response
+
+                if items_key_path:
+                    raw_item = dict_safe_get(raw_item, items_key_path)
+
+                if raw_item is None:
+                    break
+
+                raw_responses.append(raw_response)
+                raw_items += raw_item
+
+                # Calculate the offset and limit for the next run.
+                received_items = len(raw_item)
+                remaining_items -= received_items  # type: ignore[operator]
```

Comment:
I've changed the code a bit so it won't appear here, but basically the warning is because `remaining_items` might be `None`, although this scenario isn't possible.

Now I have 2 `type: ignore` in the decorator and they are each for the same warning ( a value that might be `None` even though it is within an `if` statement that promises it isn't).

I'd like to know your opinion on these cases, should I added a redundant `x is not None` even though this statement is saved withing a variable, for example: `is_automatic`. Or am I using `type: ignore` correclty?

Reply:
why is the type `Optional[int]` if it's not possible? Can we make it just `int`?

Reply:
yes, please add this check. It's redundant indeed, but preferred over muting mypy.

Reply:
No problem, added an `if` to the two areas that had this mypy error.